### PR TITLE
Add host.lima.internal to /etc/hosts

### PR DESCRIPTION
**Pull request by jandubois**

### CODE REVIEW

File: pkg/cidata/cidata.TEMPLATE.d/lima.env [hunk 1]
```diff
@@ -1,6 +1,7 @@
 LIMA_CIDATA_USER={{ .User }}
 LIMA_CIDATA_UID={{ .UID }}
 LIMA_CIDATA_MOUNTS={{ len .Mounts }}
+LIMA_SLIRP_GATEWAY={{ .SlirpGateway }}
```
Comment (AkihiroSuda) [inline pkg/cidata/cidata.TEMPLATE.d/lima.env:4]: Should be documented in internal.md

File: pkg/cidata/cidata.TEMPLATE.d/boot/03-etc-hosts.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -eux -o pipefail
+
+sed -i '/host.lima.internal/d' /etc/hosts
+echo -e "${LIMA_SLIRP_GATEWAY}\thost.lima.internal" >>/etc/hosts
```
Comment (AkihiroSuda) [inline pkg/cidata/cidata.TEMPLATE.d/boot/03-etc-hosts.sh:5]: Should be documented in network.md

File: pkg/cidata/cidata.TEMPLATE.d/lima.env (hunk 1 above)
Comment (jandubois) [reply pkg/cidata/cidata.TEMPLATE.d/lima.env]: Documented. I also changed the name to `LIMA_CIDATA_SLIRP_GATEWAY` for consistency.

File: pkg/cidata/cidata.TEMPLATE.d/boot/03-etc-hosts.sh (hunk 2 above)
Comment (jandubois) [reply pkg/cidata/cidata.TEMPLATE.d/boot/03-etc-hosts.sh]: Documented

File: docs/internal.md [hunk 3]
```diff
@@ -107,3 +107,4 @@ The volume label is "cidata", as defined by [cloud-init NoCloud](https://cloudin
 - `LIMA_CIDATA_MOUNTS_%d_MOUNTPOINT`: the N-th mount point of Lima mounts (N=0, 1, ...)
 - `LIMA_CIDATA_CONTAINERD_USER`: set to "1" if rootless containerd to be set up
 - `LIMA_CIDATA_CONTAINERD_SYSTEM`: set to "1" if system-wide containerd to be set up
+- `LIMA_CIDATA_SLIRP_GATEWAY`: set to the IP address of the host on the SLIRP network. `192.168.5.15`.
```
Comment (AkihiroSuda) [inline docs/internal.md:110]: 192.168.5.2

File: docs/network.md [hunk 4]
```diff
@@ -6,15 +6,15 @@ By default Lima only enables the user-mode networking aka "slirp".
...
 
 This IP address is not accessible from the host by design.
 
-Use `vde_vmnet` to allow accessing the guest IP from the host and other guests.
+Use [vde_vmnet](https://github.com/lima-vm/vde_vmnet) to allow accessing the guest IP from the host and other guests.
 
 ### Host IP (192.168.5.2)
 
-The loopback addresses of the host is accessible from the guest as 192.168.5.2.
+The loopback addresses of the host is `192.168.5.15` and is accessible from the guest as `host.lima.internal`.
```
Comment (AkihiroSuda) [inline docs/network.md:17]: 192.168.5.2

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
typo

File: docs/network.md (hunk 4 above)
Comment (jandubois) [reply docs/network.md]: oops

**Review by AkihiroSuda — APPROVED** [review]
Thanks.

After merging this, I'll move the repo to lima-vm and release v0.6.0

**Comment by jcw-** [comment]
@jandubois @AkihiroSuda fyi, I struggled a bit trying to use this from an nginx container, as I thought I would be able to use `host.lima.internal` in the configuration (under Docker I had been using `host.docker.internal`) - but it's actually only available to the lima VM itself and not available to containers it seems like? I was able to get it to work by using the actual IP of the host loopback directly though...

Screenshot below shows how it's not a meaningful host name in the container:
image

**Comment by AkihiroSuda** [comment]
Opened https://github.com/containerd/nerdctl/issues/355 for tracking the issue above

**Pull request merged**